source/npm/util.js

```diff
@@ -100,12 +102,65 @@ exports.verifyRecentNpmVersion = async () => {
 };
 
 exports.checkIgnoreStrategy = ({files}) => {
-	const rootDir = pkgDir.sync();
-	const npmignoreExists = fs.existsSync(path.resolve(rootDir, '.npmignore'));
-
-	if (!files && !npmignoreExists) {
+	if (!files && !npmignoreExistsInPackageRootDir()) {
 		console.log(`
 		\n${chalk.bold.yellow('Warning:')} No ${chalk.bold.cyan('files')} field specified in ${chalk.bold.magenta('package.json')} nor is a ${chalk.bold.magenta('.npmignore')} file present. Having one of those will prevent you from accidentally publishing development-specific files along with your package's source code to npm. 
 		`);
 	}
 };
+
+// Get all files which will be ignored by either `.npmignore` or the `files` property in `package.json` (if defined)
+exports.getNewAndUnpublishedFiles = async (newFiles, globArrayFromFilesProperty) => {
+	if (globArrayFromFilesProperty) {
+		return getFilesNotIncludedInTheFilesProperty(globArrayFromFilesProperty, newFiles);
+	}
+
+	if (npmignoreExistsInPackageRootDir()) {
+		return getFilesIgnoredByDotnpmignore(newFiles);
+	}
+};
+
+function npmignoreExistsInPackageRootDir() {
+	const rootDir = pkgDir.sync();
+	return fs.existsSync(path.resolve(rootDir, '.npmignore'));
+}
+
+async function getFilesIgnoredByDotnpmignore(fileList) {
+	const whiteList = await ignoreWalker({
+		path: pkgDir.sync(),
+		ignoreFiles: ['.npmignore']
+	});
+	return fileList.filter(minimatch.filter(getIgnoredFilesGlob(whiteList),
+		{matchBase: true}));
+}
+
+function getFilesNotIncludedInTheFilesProperty(globArrayFromFilesProperty, fileList) {
+	const globArrayForFilesAndDirs = [...globArrayFromFilesProperty];
+	const rootDir = pkgDir.sync();
+	for (const glob of globArrayFromFilesProperty) {
+		try {
+			if (fs.statSync(path.resolve(rootDir, glob)).isDirectory()) {
+				globArrayForFilesAndDirs.push(`${glob}/**/*`);
+			}
+		} catch (_) {}
+	}
+
+	return fileList.filter(minimatch.filter(getIgnoredFilesGlob(globArrayForFilesAndDirs),
+		{matchBase: true}));
+}
+
+function getIgnoredFilesGlob(globArrayFromFilesProperty) {
+	const filesIgnoredByDefault = ['.*.swp',
```

Comment:
Maybe add a reference comment indicating why these files are ignored by default, and the source you got this from?

Reply:
I added a comment and i checked npm-doc again. The files `.wafpickle-N`, `*.orig`, `package-lock.json` and `node_modules` are ignored by default and weren't included in the last commit. Now i added them.